Source/WTF/wtf/text/icu/UTextProviderLatin1.cpp

```diff
@@ -184,7 +184,7 @@ static int32_t uTextLatin1Extract(UText* uText, int64_t start, int64_t limit, UC
         if (trimmedLength > dest.size())
             trimmedLength = dest.size();
 WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
-        StringImpl::copyCharacters(dest.data(), unsafeMakeSpan(static_cast<const LChar*>(uText->context) + start, trimmedLength));
+        StringImpl::copyCharacters(dest, unsafeMakeSpan(static_cast<const LChar*>(uText->context) + start, trimmedLength));
```

Comment:
If we had a function that takes a `UText` and returns a `span` then this would be using `subspan`.

Reply:
I'm not familiar with ICU and UText is still a bit confusing to me. It seems UText has both `chunkContents` and `context`. Converting a UText to a span for `chunkContents` seems easy enough because I can rely on `chunkContents`  + `chunkLength` I think.

However, this line is about `context` which is a `void*` and I'm not sure what it corresponds to or its size.

Reply:
Note that this code is still behind `WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN` and will have to be re-visited no matter what. I can use `std::span { }` instead of `unsafeMakeSpan()` for clarity.

Reply:
Yes, `chunkContents` and `chunkLength` go hand in hand and point to the current chunk. So it’s great to have the `chunkSpan` function that takes a `UText`.

Since this is the implementation of a `UText` provider, `UTextProviderLatin1`. As the provider, we can store whatever pointer we want in `context`, and it’s effectively the pointer to “all the text” in a provider-specific sense. And so we should have a function specific to `UTextProviderLatin1` that makes a `span` that combines `context` with `a`. This also takes a `UText`, but needs to be a `UTextProviderLatin1`. Could assert `pFuncs == uTextLatin1Funcs` .